pkg/identity/persistence.go

```diff
@@ -0,0 +1,57 @@
+package identity
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/jackc/pgx/v4"
+)
+
+func dbConn(c *gin.Context) (*pgx.Conn, bool) {
+	conn, ok := c.MustGet("conn").(*pgx.Conn)
+	return conn, ok
+}
+
+// UserInDB Checks if the User is in the database.
+func UserInDB(c *gin.Context, conn *pgx.Conn, email string) (exists bool, err error) {
+
+	query := `
+        SELECT
+            count(*) > 0
+        FROM
+            users
+        WHERE
+            email=$1
+    `
+	err = conn.QueryRow(
+		context.Background(), query, email,
+	).Scan(&exists)
+
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"detail": "please try again later",
+		})
+		return
```

Comment:
This return can be removed as the one directly below it will be hit after anyway.